src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  return sourceString
+    .split(';')
+    .map((style) => style.trim())
+    .filter((style) => style.length > 0)
+    .reduce((stylesObject, style) => {
+      const [property, value] = style.split(':').map((part) => part.trim());
```

Comment:
[CODE KNOWLEDGE] - It's good that you're using `map` to trim both the property and value parts of the style declaration. However, ensure that the value is defined before assigning it to the `stylesObject` to avoid setting undefined properties.